pkg/daemon/update.go

```diff
@@ -356,7 +352,8 @@ func (dn *Daemon) applyOSChanges(oldConfig, newConfig *mcfgv1.MachineConfig) (re
 		if dn.recorder != nil {
 			dn.recorder.Eventf(getNodeRef(dn.node), corev1.EventTypeNormal, "InClusterUpgrade", fmt.Sprintf("Updating from oscontainer %s", newConfig.Spec.OSImageURL))
 		}
-		if osImageContentDir, err = ExtractOSImage(newConfig.Spec.OSImageURL); err != nil {
+		osImageContentDir, err := ExtractOSImage(newConfig.Spec.OSImageURL)
```

Comment:
This is not idiomatic in go. You can leave the way it was earlier. Also, osImageContentDir var has already been created earlier in the function, so you are limiting the scope of this var by recreating a local scoped variable.

Reply:
Redefined `err` instead